soundbay/data.py

```diff
@@ -232,6 +231,46 @@ def __len__(self):
         return self.metadata.shape[0]
 
 
+
+
+class MultiClassifierDataset(ClassifierDataset):
+    '''
+    This  class inherits all the traits from ClassifierDataset and handles cases with multiple classes
+    '''
+
+    
+    def _get_audio(self, path_to_file, begin_time, end_time, label):
+        """
+        _get_audio gets a path_to_file from _grab_fields method and also begin_time and end_time
+        and returns the audio segment in a torch.tensor. it differs from ClassifierDataset _get_audio method
+        by turning the first conditional into "not background" instead of "if call"
+
+        input:
+        path_to_file - string
+        begin_time - int
+        end_time - int
+
+        output:
+        audio - pytorch tensor (1-D array)
+        """
+        
+
+        if (self.mode == "train") and (label != 0):
+            if self.margin_ratio != 0:  # ranges from 0 to 1 - indicates the relative part from seq_len to exceed call_length
+                margin_len_begin = int((self.seq_length * self.data_sample_rate) * self.margin_ratio)
```

Comment:
How do we avoid getting other non background signals inside our margins?

Reply:
yes, at least for the "only calls" setup where we don't have a specified background call